File: Problems/No.49_Group_Anagrams/group_anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
     
        // parameters
        vector<vector<string>> Anagrams;
        map<string, vector<string>> mapping;
        string temp;
        
        // start compute
        for (auto str : strs) {
            
            temp = str;
            sort(temp.begin(), temp.end());
            mapping[temp].push_back(str);
            
        }
        for (auto ele : mapping) Anagrams.push_back(ele.second);
        
        return Anagrams;
        
    }
};

File: Problems/No.49_Group_Anagrams/group_anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        
        # parameter
        Anagrams = defaultdict(list)
        
        # start compute
        for s in strs:
            Anagrams[''.join(sorted(s))].append(s)
        
        return Anagrams.values()
